tests/integration/relationships/has-many-test.js

```diff
@@ -1077,6 +1077,26 @@ test("Polymorphic relationships with a hasMany is set up correctly on both sides
   assert.equal(get(email, 'posts.length'), 1, "The inverse has many is set up correctly on the email side.");
 });
 
+test("Can push a polymorphic belongsTo, the relationship is set up correctly on both sides", function(assert) {
+  assert.expect(2);
+
+  Comment.reopen({
+    message: DS.belongsTo('message', { polymorphic: true, async: false })
+  });
+  var post, comment;
+
+  run(function () {
+    post = env.store.push({ data: { type: 'post', id: 1 } });
```

Comment:
Looks like this JSONAPI document is missing some data about the `comment` relationship.